hub/powertoys/command-palette/publish-extension.md

```diff
@@ -1,72 +1,585 @@
+    Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" <!-- Replace with Package/Identity/Publisher -->
+    Version="0.0.1.0" />
+
+  <Properties>
+    <DisplayName>TemplateDisplayName</DisplayName> <!-- Replace with the reserved name -->
+    <PublisherDisplayName>A Lone Developer</PublisherDisplayName> <!-- Replace with Package/Properties/PublisherDisplayName -->
+    <Logo>Assets\StoreLogo.png</Logo> <!-- Update -->
+  </Properties>
+```
+
+1. In your IDE, open `<ExtensionName>.csproj`.
+1. Locate the `PropertyGroup` element and add the following properties:
+
+```xml
+    <AppxPackageIdentityName>Package/Identity/Name</AppxPackageIdentityName>
+    <AppxPackagePublisher>CN=########-####-####-####-############</AppxPackagePublisher>
+    <AppxPackageVersion>0.0.1.0</AppxPackageVersion>
+```
+
+### Build MSIX
+
+1. In the terminal, move to the `<ExtensionName>\<ExtensionName>` directory.
+1. Create an x64 build MSIX with the following command:
+
+   ```powershell
+   dotnet build --configuration Release -p:GenerateAppxPackageOnBuild=true -p:Platform=x64
+   ```
 
-The following tools are required to build and publish your extension:
+1. Create an ARM64 build MSIX with the following command:
 
-- [Visual Studio 2022](https://visualstudio.microsoft.com/vs/) (Community, Professional, or Enterprise edition)
+   ```powershell
+   dotnet build --configuration Release -p:GenerateAppxPackageOnBuild=true -p:Platform=ARM64
+   ```
+
+1. Locate the MSIX files:
+
+   ```powershell
+   cd <ExtensionName>; dir bin -Recurse -Filter "*.msix"
+   ```
+
+1. Note the locations of the `<ExtensionName>_<VersionNumber>_x64.msix` and `<ExtensionName>_<VersionNumber>_arm64.msix` files.
+
+1. Create a new `bundle_mapping.txt` file and include the following content, updating the paths to your MSIX files:
+
+   ```text
+   [Files]
+   "bin\x64\Release\PATH\<ExtensionName>_<VersionNumber>_x64.msix" "<ExtensionName>_<VersionNumber>_x64.msix"
+   "bin\ARM64\Release\PATH\<ExtensionName>_<VersionNumber>_arm64.msix" "<ExtensionName>_<VersionNumber>_arm64.msix"
+   ```
+
+1. Create a bundle that combines both architectures into a single package for Microsoft Store submission. Update the `<ExtensionName>` and `<VersionNumber>`:
+
+   ```powershell
+   makeappx bundle /v /d bin\Release\ /p <ExtensionName>_<VersionNumber>_Bundle.msixbundle
+   ```
+
+   > [!NOTE]
+   > If `makeappx` isn't recognized, check which version of Windows SDK you have installed:
+   >
+   > ```powershell
+   > Get-ChildItem "HKLM:\SOFTWARE\WOW6432Node\Microsoft\Microsoft SDKs\Windows\v*" | Select-Object PSChildName
+   > ```
+   >
+   > Then update the following script with the version number:
+   >
+   > ```powershell
+   > & "C:\Program Files (x86)\Windows Kits\<VersionNumber>\bin\[version]\x64\makeappx.exe" bundle /f bundle_mapping.txt /p <ExtensionName>_<VersionNumber>_Bundle.msixbundle
+   > ```
```

Comment:
Think there needs to be a bit more clarity here on what these version numbers should be:
The first command returned `v10.0` for me, but the expected path val was `10`

I also assumed that those two version values were the same because the command returns just one - but my SDK version was `10.0.26100.0`

My full path ended up being `"C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\arm64\makeappx.exe` which might be hard to gather from these instructions.